src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -233,21 +371,128 @@ void ElementView::deleteSelectedItems()
     doc->commitTransaction();
 }
 
+void ElementView::onMouseHoverIndex(QModelIndex index) {
+    Q_EMIT onMouseHoverItem(itemFromIndex(index));
+}
 
-void ElementView::keyPressEvent(QKeyEvent * event)
+ElementItem* ElementView::itemFromIndex(const QModelIndex& index) {
+    return static_cast<ElementItem*>(QListWidget::itemFromIndex(index)); //required because QListWidget::itemFromIndex is protected.

```

Comment:
elementView->itemFromIndex(index) in ElementItemDelegate wasn't working initially because that function is protected. So I had to reimplementate itemFromIndex as a public member of ElementView.
The comment is useless I'll remove it.